lib/rubygems/version.rb

```diff
@@ -170,6 +170,8 @@ def version
   # True if the +version+ string matches RubyGems' requirements.
 
   def self.correct? version
+    warn "nil versions are discouraged and will be deprecated in Rubygems 4" if version.nil?
+
     return false if version.nil?
```

Comment:
Remove this line for reverting previous behavior.

Reply:
done